### PR TITLE
fix: Warning user to install signatures

### PR DESCRIPTION
**Pull request by naikordian**
Give the user an instruction link to install signatures if they installed Capa via pip
related #682

### Checklist

- [x] No CHANGELOG update needed

- [x] No new tests needed

- [x] No documentation update needed

### CODE REVIEW

File: capa/main.py [hunk 1]
```diff
@@ -699,6 +699,12 @@ def get_rules(
 
 def get_signatures(sigs_path):
     if not os.path.exists(sigs_path):
+        # Check if Capa installed via PIP
+        if "/site-packages/capa/" in sigs_path:
```
Comment (williballenthin) [inline capa/main.py:703]: this check won't work on windows because of the path separator differences

**Comment by williballenthin** [comment]
i like the idea of providing more feedback to the user when the signatures can't be found, thank you for proposing these changes. see above for some suggestions.

**Comment by naikordian** [comment]
I've made the change you recommended and moved the check closer to the CLI args. This will now issue a warning if the default path is used but the directory doesn't exist. However, I'm not certain if I should also check if the signature path is within a pip package or not.

File: capa/main.py [hunk 2]
```diff
@@ -698,9 +698,6 @@ def get_rules(
 
 
 def get_signatures(sigs_path):
-    if not os.path.exists(sigs_path):
-        raise IOError(f"signatures path {sigs_path} does not exist or cannot be accessed")
-
```
Comment (williballenthin) [inline capa/main.py:703]: we should still raise this exception if the path doesn't exist.

in the changes below, there's a check and error if the default path is used, but when the user provides `--signatures=/does/not/exist` which should still be handled here.

File: capa/main.py [hunk 3]
```diff
@@ -1038,6 +1035,11 @@ def handle_common_args(args):
             logger.debug("-" * 80)
 
             sigs_path = os.path.join(get_default_root(), "sigs")
+            if not os.path.exists(sigs_path):
+                raise IOError(
+                    "Please install the signatures first: "
+                    "https://github.com/mandiant/capa/blob/master/doc/installation.md#method-2-using-capa-as-a-python-library."
+                )
```
Comment (williballenthin) [inline capa/main.py:1042]:
```suggestion
            if not os.path.exists(sigs_path):
                logger.error(
                    "Using default signature path, but it doesn't exist. Please install the signatures first: "
                    "https://github.com/mandiant/capa/blob/master/doc/installation.md#method-2-using-capa-as-a-python-library."
                )
                raise IOError(f"signatures path {sigs_path} does not exist or cannot be accessed")
```

use the logger to show messages to the user, while the exception should contain the problem itself.

**Review by williballenthin — CHANGES_REQUESTED** [review]
see two recommendations inline, then ready to merge

File: capa/main.py (hunk 2 above)
Comment (naikordian) [reply capa/main.py]: Fixed

**Comment by mr-tz** [comment]
Looks good, can you please rebase (to fix the black code_style fail)?

**Comment by mr-tz** [comment]
@williballenthin g2g?

**Comment by williballenthin** [comment]
send it!Message ID: ***@***.***>
thanks @naikordian!

**Pull request merged**